source/MaterialXTest/ShaderValid.cpp

```diff
@@ -303,22 +309,43 @@ static mx::GlslValidatorPtr createValidator(bool& orthographicView, const std::s
 //
 // Outputs error log if validation fails
 //
-static void runValidation(const std::string& outputPath, const std::string& shaderName, mx::ElementPtr element,
-                          mx::GlslValidatorPtr validator, mx::ShaderGeneratorPtr shaderGenerator,
-                          bool orthographicView, mx::DocumentPtr doc, std::ostream& log, bool outputMtlxDoc=false)
+static void runValidation(const std::string& shaderName, mx::ElementPtr element, mx::GlslValidatorPtr validator,
+                          mx::ShaderGeneratorPtr shaderGenerator, bool orthographicView, mx::DocumentPtr doc,
+                          std::ostream& log, bool outputMtlxDoc=true, const std::string& outputFolder="",
+                          const std::string& outputPath=".")
```

Comment:
I'm curious why do we need an output folder + an output path. Can't we just have an output path and make the default something like "./results" ?

Reply:
Not sure what you mean, if I have a test running called math.mtlx this generates a math folder to put the results in. A results folder wouldn't make sense as it would then place the results of all tests in the same folder into the same results folder which seems messy to me.

Reply:
Currently the results are put in the same folder where the source file is. e.g. "documents/stdlib/math/"
 as "." (current directory) is used as the output directory. If you set it to be "./results" result files will be in a sub-folder relative to the current directory. In this case: "documents/stdlib/math/results". 

As for 1 vs 2 args, I see no need to specify both path + folder to build the final output location.